### PR TITLE
Feature: Item Card / Markdown

### PR DESCRIPTION
**Pull request by moiskillnadne**
Shared components:
- markdown
- TextItem
- progress bar

Pages:
- Activity details

Widgets:
- Activity Details
- Activity Progress Preview
-

### CODE REVIEW

File: src/entities/activity/ui/ActivityProgressPreviewList.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { ActivityListItem } from "../lib"
+import { getRandomInt } from "../lib/getRandomInt"
+import { ActivityProgressPreviewCard } from "./ActivityProgressPreviewCard"
+
+interface ActivityProgressPreviewListProps {
+  activities: ActivityListItem[]
+}
+
+export const ActivityProgressPreviewList = ({ activities }: ActivityProgressPreviewListProps) => {
+  const activitiesPreview = activities.map(activity => {
```
Comment (BamMironov) [inline src/entities/activity/ui/ActivityProgressPreviewList.tsx:10]: Please wrap it in useMemo

File: src/entities/activity/ui/ActivityProgressPreviewList.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const itemsLength = 10 // activity.items.length in the real implementation
```
Comment (BamMironov) [inline src/entities/activity/ui/ActivityProgressPreviewList.tsx:11]: Also, please move the mapper function to the `model` level

File: src/entities/item/ui/CardItemNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  isSubmitButtonShown?: boolean
+  isNextButtonShown?: boolean
+  onNextButtonClick?: () => void
+
+  isSkippable?: boolean
+  isNextButtonDisable?: boolean
+}
+
+export const CardItemNavigator = ({
```
Comment (BamMironov) [inline src/entities/item/ui/CardItemNavigator.tsx:18]: You do not need this component. Just use BasicButton component where is makes sense

File: src/pages/index.tsx [hunk 4]
```diff
@@ -69,6 +69,14 @@ const ApplicationRouter = (): JSX.Element | null => {
         </Route>
       </Route>
 
+      <Route path="/test">
```
Comment (BamMironov) [inline src/pages/index.tsx:72]: Please remove `/` sign from here

File: src/shared/api/types/item.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+export interface Item {
```
Comment (BamMironov) [inline src/shared/api/types/item.ts:1]: Please come up with a different name

File: src/widgets/ActivityDetails/ui/BackNavigateButton.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+import { BasicButton } from "~/shared/ui"
+import { useCustomTranslation } from "~/shared/utils"
+
+export const BackNavigateButton = () => {
+  const { t } = useCustomTranslation()
+  const navigate = useNavigate()
+
+  const onBackButtonClick = () => {
+    navigate(-1)
```
Comment (BamMironov) [inline src/widgets/ActivityDetails/ui/BackNavigateButton.tsx:12]: Please add a guard in case when there is no way previous page

Comment (BamMironov) [reply src/widgets/ActivityDetails/ui/BackNavigateButton.tsx:12]: And if there is no previous page make sure to provide a "falback" route

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/entities/activity/ui/ActivityProgressPreviewList.tsx (hunk 1 above)
Comment (moiskillnadne) [reply src/entities/activity/ui/ActivityProgressPreviewList.tsx:10]: Done

File: src/entities/activity/ui/ActivityProgressPreviewList.tsx (hunk 2 above)
Comment (moiskillnadne) [reply src/entities/activity/ui/ActivityProgressPreviewList.tsx:11]: Done

File: src/entities/item/ui/CardItemNavigator.tsx (hunk 3 above)
Comment (moiskillnadne) [reply src/entities/item/ui/CardItemNavigator.tsx:18]: Done

File: src/pages/index.tsx (hunk 4 above)
Comment (moiskillnadne) [reply src/pages/index.tsx:72]: Remove this part entirely, thank you. Done

File: src/shared/api/types/item.ts (hunk 5 above)
Comment (moiskillnadne) [reply src/shared/api/types/item.ts:1]: Done

File: src/widgets/ActivityDetails/ui/BackNavigateButton.tsx (hunk 6 above)
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/ui/BackNavigateButton.tsx:12]: Done, fallback no need here

**Review by BamMironov — APPROVED** [review]

**Pull request merged**